common/headtracker/types/head.go

```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+//go:generate mockery --quiet --name Head --output ./mocks/ --case=underscore
+type Head[H types.Head[BLOCK_HASH], BLOCK_HASH types.Hashable, CHAIN_ID txmgrtypes.ID] interface {
+	types.Head[BLOCK_HASH]
+	// Equals returns true if the two heads are equal
+	Equals(H) bool
+	// ChainID returns the chain ID that the head is for
+	ChainId() CHAIN_ID
+	// Check if the chain ID of the Head is nil
+	IsChainIdNil() bool
+	// Check if the two heads are of the same chain
+	IsSameChain(CHAIN_ID) bool
```

Comment:
This seems odd - is there something in the way of comparing `ChainID()`s?

Reply:
The main issue is that for Evm chains, the chain ids are in `big.Int` . However for Non EVM chains like solana, the Ids are in `string` . So i thought that the generalised approach would be to use a method to check if the chain IDs are the same. 

This function is needed over here 
https://github.com/smartcontractkit/chainlink/blob/ebc1929487248412e59648931c396a81fe29c288/core/chains/evm/headtracker/head_listener.go#L169

To check for the edge case that the client and head are of different chain.

Reply:
Or the strings could be compared, so the interface api doesn't have to expand:
`blockHeader.ChainID().String() == chainId.String()`

Reply:
Good point. My main issue is that by implementing `Equals` for type `ID`, it would result in all `Chain_ID` of type `ID` to have either `Equals` or `Cmp` Methods. This could propagate deep into our codebase especially when that our EVM code base is using `*big.Int` as Chain_ID. 

I do think that `blockHeader.ChainID().Equals(chainId)` has merits in readability, but is there someway of naming the method to compare `ChainId` without making all ID implement Equals method?

For instance - Does `CmpChainId` or `HasSameId` work?

Reply:
Or do you think it is better to have ID type interface to have comparison methods?

something like

```
type ID interface {
	fmt.Stringer

	Cmp(fmt.Stringer) int
}
```

Or 

```
type ID[O comparator] interface {
	fmt.Stringer

	Cmp(O) int
}
```

Reply:
I think that string comparison or requiring `comparable` are more appealing than `Cmp` or `Equals`. Only string comparison avoids cascading changes though, so that seems simplest. Only tradeoff is efficiency?

Reply:
yup agreed! string comparison suggestion is great - instead of having a method for `Head` that compares `ChainId`, we could use a string comparison for 

https://github.com/smartcontractkit/chainlink/blob/40e7344f903ea4fdc672ac2ff27aa21c5b954e03/core/chains/evm/headtracker/head_listener.go#L170

Reply:
Updated to remove the function